aries_cloudagent/registrar/default/indy.py

```diff
@@ -138,69 +136,95 @@ async def create(
-            try:
-                transaction = await transaction_mgr.create_record(
-                    messages_attach=txn["signed_txn"],
-                    connection_id=connection_id,
-                    meta_data=meta_data,
+    async def create(
+        self,
+        profile: Profile,
+        method: Optional[str],
+        did: Optional[str] = None,
+        document: Optional[dict] = None,
+        **options: dict,
+    ) -> JobRecord:
+        """Create a DID from a given method."""
+        raise NotImplementedError
+        responder = profile.inject(BaseResponder)
+        # TODO Add multi-ledger suppport through (pseudo) did indy method support
+        async with profile.session() as session:
+            ledger = session.inject_or(BaseLedger)
+            await self._check_ledger(ledger, session.settings.get_value("wallet.type"))
+            # Create DID
+            did_info = await self._create_did(session, method, options)
+
+            # TODO how should this interact with optional did param?
+            did, verkey = did_info.did, did_info.verkey
+
+            alias = options.get("alias")
+            role = options.get("role")
+            if role == "reset":  # indy: empty to reset, null for regular user
+                role = ""  # visually: confusing - correct 'reset' to empty string here
+            author = is_author(profile)
+            if author:
+                endorser_connection_id = await self._retreive_endorsor_connection_id()
+                endorser_did = self._retreive_endorsor_did(profile, endorser_connection_id)
+
+            (success, txn) = self._register_nym(ledger, did, verkey, alias, role, not author, endorser_did)
+            
+            meta_data = {"verkey": verkey, "alias": alias, "role": role}
+            if author:
+                _endorser, endorsement = self._endorse_txn(profile, endorser_connection_id, txn, meta_data)
+                # if auto-request, send the request to the endorser
+                if profile.settings.get_value("endorser.auto_request"):
+                    try:
+                        (
+                            endorsement,
+                            transaction_request,
+                        ) = await _endorser.create_request(
+                            transaction=endorsement,
+                            # TODO see if we need to parameterize these params
+                            # expires_time=expires_time,
+                            # endorser_write_txn=endorser_write_txn,
+                        )
+                    except (StorageError, TransactionManagerError) as err:
+                        raise RegistrarError(err.roll_up) from err
+
+                    await responder.send(transaction_request, connection_id=endorser_connection_id)
+            else:
+                await profile.notify(
+                    DID_CREATION_TOPIC + did,
+                    meta_data,
```

Comment:
It's in the git history, we should delete this instead of leaving the dead code around.

Reply:
Yes, I will clean up the code.